port/board/afck/cli_commands.c

```diff
@@ -1,700 +1,854 @@
+                /* Undo truncation to 1 byte and take into account that 1 LSB is 4 mV */
+                printf("Sensor %s : %5d mV\n", sdr_entry->IDstring, sensor->readout_value << 6 );
+            } else if (sdr_entry->sensortype == SENSOR_TYPE_CURRENT) {
+                /* Undo truncation and print signed */
+                printf("Sensor %s : %4d mA\n", sdr_entry->IDstring, (int16_t) (sensor->readout_value << 5));
+            }
+        }
+        if (repeat) {
+            if (uart_read(UART_DEBUG, pcWriteBuffer, 1)) {
+                repeat = false;
+                /* Discard byte that was read */
+                pcWriteBuffer[0] = '\0';
+            } else {
+                uart_send(UART_DEBUG, cursor_restore, sizeof(cursor_restore)-1);
+                vTaskDelay(100 / portTICK_PERIOD_MS);
+            }
+        }
+    } while (repeat);
+    return pdFALSE;
+}
+
+static BaseType_t clockConfigReadCommand(char *pcWriteBuffer, size_t xWriteBufferLength,
+                                         const char *pcCommandString) {
+    // Initialize the output buffer with an empty string
+    pcWriteBuffer[0] = '\0';
+
+    // Define the number of clocks to configure
+    const uint8_t numberOfClocks = 16;
+
+    // Initialize variables to store the extracted enable mask and output map
+    uint16_t enableMaskValue = 0;
+    uint64_t outputMapValue = 0;
+
+    // Print the clock configuration table header
+    printf("| Output          | Enabled | Input                   |\r\n");
+    printf("|-----------------|---------|-------------------------|\r\n");
+
+    // Iterate over each clock output to read the current configuration
+    for (uint8_t i = 0; i < numberOfClocks; ++i) {
+        // Read the configuration for the current clock output
+        uint8_t config = clock_config[i];
+
+        // Extract the enable bit from the 7th position and
+        // update the enable mask by setting the corresponding bit
+        uint8_t enableOutput = (config >> 7) & 0x1;
+        enableMaskValue |= (enableOutput << i);
+
+        // Extract the 4-bit selected input source from the lower bits and
+        // update the output map by setting the corresponding 4-bit value
+        uint8_t selectedInput = config & 0xF;
+        outputMapValue |= ((uint64_t)selectedInput << (4 * i));
+
+        // Print the current clock configuration
+        printf("| %-2u %-12s | %-7u | %-2u %-20s |\r\n", i, ADN4604_OUTPUT[i],
+               enableOutput, selectedInput, ADN4604_INPUT[selectedInput]);
+    }
+
+    // Print the current enable mask
+    printf("\r\nCurrent enable mask: 0x%04X\r\n", enableMaskValue);
+
+    // Print the current clock output map
+    printf("Current clock output map: 0x");
+    for (uint8_t i = numberOfClocks - 1; i < numberOfClocks; i--) {
+        // Extract each 4-bit segment and print as a hexadecimal digit
+        uint8_t segment = (outputMapValue >> (4 * i)) & 0xF;
+        printf("%X", segment);
```

Comment:
Can't you just print `outputMapValue` as `%016X` ?

Reply:
You would have to add `printlli`, similar to https://github.com/tiqi-group/openMMC/blob/feature/afck/modules/printf-stdarg.c#L127 (and adapt `print` accordingly)

But there is no real reason to use a custom printf implementation. 
https://github.com/eyalroz/printf/ would be a better alternative.

At least write a comment why you did this.